src/Chainweb/Rosetta/Internal.hs

```diff
@@ -118,73 +108,90 @@ matchLogs
     -> M.Map TxId [AccountLog]
     -> CoinbaseTx (CommandResult Hash)
     -> V.Vector (CommandResult Hash)
-    -> Either RosettaFailure tx
+    -> ExceptT RosettaFailure Handler tx
 matchLogs typ bh logs coinbase txs
   | bheight == genesisHeight v cid = matchGenesis
-  | coinV2Upgrade v cid bheight = matchRemediation
+  | coinV2Upgrade v cid bheight = matchRemediation upgradeTransactions
+  | to20ChainRebalance v cid bheight = matchRemediation twentyChainUpgradeTransactions
```

Comment:
TODO: add coinv3 upgrade txs

Reply:
Done, going to run rosetta-cli tests soon

Reply:
But did add local coinv3 block test